jazon-core/src/test/groovy/com/zendesk/jazon/MatcherSpec.groovy

```diff
@@ -423,6 +427,41 @@ class MatcherSpec extends Specification {
         thrown(IllegalStateException)
     }
 
+    @Unroll
+    def "array each element expectation: success"() {
+        expect:
+        match([a: anyNumberOf(expected)], [a: actual]).success()
+
+        where:
+        expected                                       | actual
+        '1'                                            | []
+        '1'                                            | ['1']
+        '1'                                            | ['1', '1']
+        true                                           | [true]
+        2                                              | [2]
+        { it -> it > 5 } as Predicate<Integer>  | [6, 7, 8]
+    }
+
+    @Unroll
+    def "array each element expectation - element mismatch"() {
+        when:
+        def result = match([a: anyNumberOf(expected)], [a: actual])
+
+        then:
+        !result.ok()
+        result.mismatch().expectationMismatch() == elementMismatch
+        result.mismatch().path() == '$.a.' + elementIndex
+
+        where:
```

Comment:
It would be great to add 2 additional cases: 
* for sample json-object as the repeated member of verified array
* for sample json-array as the repeated member of verified array

Reply:
Added tests for maps and lists as the expected element. Not sure if that's what you had in mind

Reply:
This one is cool :)